demos/testnet_app/webpack.config.js

```diff
@@ -0,0 +1,105 @@
+const path = require("path");
+const webpack = require("webpack");
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const TerserPlugin = require("terser-webpack-plugin");
+const dfxJson = require("./dfx.json");
+
+// List of all aliases for canisters. This creates the module alias for
+// the `import ... from "@dfinity/ic/canisters/xyz"` where xyz is the name of a
+// canister.
+const aliases = Object.entries(dfxJson.canisters).reduce(
+  (acc, [name, _value]) => {
+    // Get the network name, or `local` by default.
+    const networkName = process.env["DFX_NETWORK"] || "local";
+    const outputRoot = path.join(
+      __dirname,
+      ".dfx",
+      networkName,
+      "canisters",
+      name
+    );
+
+    return {
+      ...acc,
+      ["dfx-generated/" + name]: path.join(outputRoot, name + ".js"),
+    };
+  },
+  {}
+);
+
+/**
+ * Generate a webpack configuration for a canister.
+ */
+function generateWebpackConfigForCanister(name, info) {
+  if (typeof info.frontend !== "object") {
+    return;
+  }
+
+  return {
+    mode: "production",
+    entry: {
+      // The frontend.entrypoint points to the HTML file for this build, so we need
+      // to replace the extension to `.js`.
+      index: path.join(__dirname, info.frontend.entrypoint).replace(/\.html$/, ".js"),
+    },
+    devServer: {
+      port: 8081,
+      proxy: {
+        "/api": "http://localhost:8000"
+      },
+      allowedHosts: [".localhost", ".local", ".ngrok.io"],
+    },
+    optimization: {
+      minimize: true,
+      minimizer: [new TerserPlugin()],
+    },
+    resolve: {
+      alias: aliases,
+      extensions: [".js", ".ts", ".jsx", ".tsx"],
+      fallback: {
+        "assert": require.resolve("assert/"),
+        "buffer": require.resolve("buffer/"),
+        "events": require.resolve("events/"),
+        "stream": require.resolve("stream-browserify/"),
+        "util": require.resolve("util/"),
+      },
+    },
+    output: {
+      filename: "[name].js",
+      path: path.join(__dirname, "dist", name),
+    },
+
+    // Depending in the language or framework you are using for
+    // front-end development, add module loaders to the default
+    // webpack configuration. For example, if you are using React
+    // modules and CSS as described in the "Adding a stylesheet"
+    // tutorial, uncomment the following lines:
+    // module: {
+    //  rules: [
+    //    { test: /\.(ts|tsx|jsx)$/, loader: "ts-loader" },
+    //    { test: /\.css$/, use: ['style-loader','css-loader'] }
+    //  ]
+    // },
```

Comment:
I guess this was copy pasted, probably not worth duplicating here

Reply:
All of this is the dfx generated standard project. I left it here in the hope that this will make it easier to update for newer breaking dfx releases, since the migration instructions they put out will correspond closely to our setup.